bot.go

```diff
@@ -127,11 +130,17 @@ func (b *Bot) SendMessage(target string, message string, sender *User) {
 	}
 }
 
-func (b *Bot) messageSender() {
+func (b *Bot) sendResponse(target, message string, sender *User) {
+	b.mutex.Lock()
+	b.handlers.Response(target, message, sender)
+	b.mutex.Unlock()
+}
+
+func (b *Bot) processMessages() {
 	for {
 		select {
 		case msg := <-b.msgsToSend:
-			b.handlers.Response(msg.target, msg.message, msg.sender)
+			b.sendResponse(msg.target, msg.message, msg.sender)
```

Comment:
This isn't a race. Only one goroutine will ever be calling messageSender, so there is nothing to synchronize between.

This is an unnecessary lock and unlock.

Reply:
The passive commands run in a goroutine and can call the SendMessage at the same time. If you run `go test --race` it will point that out.

But I think that you are correct in this case, that the Response method, at least in production will not be called at the same time.

But in the tests, as they are called synchronously it is creating a race condition.

I'll think about some way to fix the tests as having the tests break becausa of a race condition that will not be in production isn't a good thing.